### ISSUE TITLE
Last song on album scrobbles twice for Last.fm

### ISSUE THREAD

**Issue by xCurlyQ** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#1614]
**OS:** 
Windows 10

**Issue Descriptions:**
Whenever an album finishes, with the last song being played completely, and you start playing another song/album, the last song on the previous album scrobbles again. 

**Steps to Reproduce:**
Listen to the last song on an album to completion. Then, start playing a new song. The last song should show being scrobbled twice.

**Comment by rjawiygvozd** [comment]
This issue still exists for me

**Comment by captainnapalm** [comment]
I've had this occur many times. It seems to scrobble the last song multiple times, even though it has completed playing. The next time it occurs I'll be sure to grab a screenshot, but I came back to work this morning and the last track that finished playing the previous day had scrobbled 10+ times.

**Comment by captainnapalm** [comment]
Has this been looked into? My playlist is complete (played a full album to completion, queue shows as empty and last track is not paused) but I'm seeing multiple scrobbles for the final song. This is just a portion of the scrobbles. They continue on the following page.

[Imgur](http://i.imgur.com/WeZNPlj.png)

**Comment by captainnapalm** [comment]
@MarshallOfSound can you please look into this? I just deleted 3 pages worth of scrobbles for a song that hadn't played in 5 hours. the queue was finished playing, but it just kept scrobbling every minute or so until I completely quit the app.

**Comment by MarshallOfSound** [comment]
No promises but I'll see if I get a change to investigate, I personally don't use last.fm

**Comment by petuhovskiy** [comment]
This issue affected me to. It looks like the problem is in the [current](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/master/src/renderer/windows/GPMWebView/playback/handler.js#L48-L66) scrobble event trigger. Scrobble happens when playback position reaches middle of the track duration(with some other conditions), but it doesn't mean that track has been played for such a time, as user can change position anytime.
Scrobbling should be done as described [on last.fm api section](http://www.last.fm/api/scrobbling).
```
A track should only be scrobbled when the following conditions have been met:

The track must be longer than 30 seconds.
And the track has been played for at least half its duration, or for 4 minutes (whichever occurs earlier.)
```
It can be easily done with keeping of playback time (when the track was in playing state). 
I implemented this in my fork and it works just fine.

**Comment by subliun** [comment]
Can confirm this is also an issue for me. The last track in an album usually scrobbles twice.

**Comment by mbolli** [comment]
Same for me. Happened a few times, had to delete hundreds of duplicate songs from last.fm.

**Comment by fernandonito** [comment]
I confirm it too. Running on Fedora 25 x86-64

![screenshot_20170304_163155](https://cloud.githubusercontent.com/assets/4334856/23581710/28e01550-00f8-11e7-80b2-495b7fc04286.png)

**Comment by MarshallOfSound** [comment]
@petuhovskiy Would probably accept that PR, just send it through and help everyone out 👍

**Comment by wadegbow** [comment]
Any word on the last.fm fixes getting merged with master? I've had a single song scrobble 100+ times when I wasn't using the app.

**Comment by captainnapalm** [comment]
Yeah, I've basically implemented my own workaround by just never letting the queue end, so I'll either always leave it paused, or just keep adding to the bottom of the queue so it never ends. It would be nice for this to be pushed out in the next version.

**Comment by heavyperil** [comment]
Also seeing the duplicate scrobbles for last tracks. Is there anything that needs to happen before #2317 can be merged?
